src/lib/tree/namespace/NamespaceMerkleTree.sol

```diff
@@ -131,4 +132,285 @@ library NamespaceMerkleTree {
 
         return namespaceNodeEquals(root, node);
     }
+
+    /// @notice Verify if contiguous elements exists in Merkle tree, given leaves, mutliproof, and root.
+    /// @param root The root of the tree in which the given leaves are verified.
+    /// @param proof Namespace Merkle multiproof for the leaves.
+    /// @param minmaxNID Namespace ID of the leaves. All leaves must have the same namespace ID.
+    /// @param data The leaves to verify. Note: leaf data must be the _entire_ share (including namespace ID prefixing).
+    /// @return `true` if the proof is valid, `false` otherwise.
+    function verifyMulti(
+        NamespaceNode memory root,
+        NamespaceMerkleMultiproof memory proof,
+        bytes8 minmaxNID,
+        bytes[] memory data
+    ) internal pure returns (bool) {
+        // Hash all the leaves to get leaf nodes.
+        NamespaceNode[] memory nodes = new NamespaceNode[](data.length);
+        for (uint256 i = 0; i < data.length; ++i) {
+            nodes[i] = leafDigest(minmaxNID, data[i]);
+        }
+
+        // Verify inclusion of leaf nodes.
+        return verifyMultiHashes(root, proof, nodes);
+    }
+
+    /// @notice Verify if contiguous leaf hashes exists in Merkle tree, given leaf nodes, multiproof, and root.
+    /// @param root The root of the tree in which the given leaf nodes are verified.
+    /// @param proof Namespace Merkle multiproof for the leaves.
+    /// @param leafNodes The leaf nodes to verify.
+    /// @return `true` if the proof is valid, `false` otherwise.
+    function verifyMultiHashes(
+        NamespaceNode memory root,
+        NamespaceMerkleMultiproof memory proof,
+        NamespaceNode[] memory leafNodes
+    ) internal pure returns (bool) {
+        uint256 leafIndex = 0;
+        NamespaceNode[] memory leftSubtrees = new NamespaceNode[](proof.sideNodes.length);
+
+        for (uint256 i = 0; leafIndex != proof.beginKey && i < proof.sideNodes.length; ++i) {
+            uint256 subtreeSize = _nextSubtreeSize(leafIndex, proof.beginKey);
+            leftSubtrees[i] = proof.sideNodes[i];
+            leafIndex += subtreeSize;
+        }
+
```

Comment:
Ah good catch, should have documented. It was left out because 1) it seemed non-critical and 2) it's actually separate functionality to verifying the multiproof. If it's desired functionality then it can be implemented as a separate function.

Reply:
makes sense :+1: